test/r401_test.js

```diff
@@ -375,6 +375,67 @@ describe('#R4 v4.0.1', () => {
   });
 });
 
+describe('#R4 v4.0.1 PatientSource meta.profile checking', () => {
+  let patientSource;
+  before(() => {
+    patientSource = cqlfhir.PatientSource.FHIRv401({
+      requireProfileTagging: true
+    });
+  });
+
+  beforeEach(() => {
+    // patientLuna has 1 Condition resource with a meta.profile set to be a US Core Condition
+    // patientJohnnie does not have meta.profile set on any resources
+    patientSource.loadBundles([patientLuna, patientJohnnie]);
+  });
+
+  afterEach(() => patientSource.reset());
+
+  it('should throw error when trying to use meta.profile with no retrieveDetails', () => {
+    const luna = patientSource.currentPatient();
+    expect(() =>
+      luna.findRecords(
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-condition-encounter-diagnosis'
+      )
+    ).to.throw();
+  });
+
+  it('should not find any resources without a matching meta.profile', () => {
+    const luna = patientSource.currentPatient();
+    const conditions = luna.findRecords('http://example.com/not-a-real-profile', {
+      datatype: '{http://hl7.org/fhir}Condition'
+    });
+    expect(conditions).to.have.length(0);
+  });
+
+  it('should find resources with matching meta.profile', () => {
+    const luna = patientSource.currentPatient();
+    const conditions = luna.findRecords(
+      'http://hl7.org/fhir/us/core/StructureDefinition/us-core-condition-encounter-diagnosis',
+      {
+        datatype: '{http://hl7.org/fhir}Condition',
+        templateId:
+          'http://hl7.org/fhir/us/core/StructureDefinition/us-core-condition-encounter-diagnosis'
+      }
+    );
+    expect(conditions).to.have.length(1);
+    expect(conditions.every(c => c.getTypeInfo().name === 'Condition')).to.be.true;
+    expect(conditions[0].meta.profile[0].value).equal(
+      'http://hl7.org/fhir/us/core/StructureDefinition/us-core-condition-encounter-diagnosis'
+    );
+  });
+
```

Comment:
Ah, shoot.  I just realized there is some functionality that you wrote which isn't currently covered by these tests.  Could you add a test like this for each version?

```js
  it('should find FHIR core resources even when they are not tagged with the core URL', () => {
    const luna = patientSource.currentPatient();
    const conditions = luna.findRecords('http://hl7.org/fhir/StructureDefinition/Condition', {
      datatype: '{http://hl7.org/fhir}Condition',
      templateId: 'http://hl7.org/fhir/StructureDefinition/Condition'
    });
    expect(conditions).to.have.length(8);
  });
```